src/models/tiles/geometry/geometry-content.ts

```diff
@@ -261,17 +266,15 @@ export const GeometryContentModel = GeometryBaseContentModel
       return self.linkedDataSets.find(ds => ds.providerId === linkedTableId);
     },
     getSelectedIds(board: JXG.Board) {
-      // returns the ids in creation order
-      return board.objectsList
-                  .filter(obj => self.isSelected(obj.id))
-                  .map(obj => obj.id);
+      return Array.from(self.metadata.selection.entries())
+        .filter(entry => entry[1]) // [0] is the ID, [1] is boolean "selected"
+        .map(entry => entry[0]);
```

Comment:
Should be able to make this a bit more self-documenting with something like:
```suggestion
        .filter(([id, selected]) => selected) // [0] is the ID, [1] is boolean "selected"
        .map(([id]) => id);
```